src/inc/quic_platform_posix.h

```diff
@@ -541,49 +540,52 @@ CxPlatPoolAlloc(
     _Inout_ CXPLAT_POOL* Pool
     )
 {
-#if DEBUG
-    if (CxPlatGetAllocFailDenominator()) {
-        return CxPlatAlloc(Pool->Size, Pool->Tag);
-    }
-#endif
     CxPlatLockAcquire(&Pool->Lock);
-    void* Entry = CxPlatListPopEntry(&Pool->ListHead);
-    if (Entry != NULL) {
-        CXPLAT_FRE_ASSERT(Pool->ListDepth > 0);
+    CXPLAT_POOL_HEADER* Header =
+    #if DEBUG
+        CxPlatGetAllocFailDenominator() ? NULL :
```

Comment:
Is this fixing a bug in the previous behavior? It looks like previously it would allocate new on true, and allocate from the pool on false. Now it's return NULL on true, and allocating from the pool or new on false.

Reply:
ok yeah, it's coming back to me now. The point of this was to skip the pool's lookaside list and always call `CxPlatAlloc` when random memory failures are enabled so that the logic in `CxPlatAlloc` can decide to return NULL.
The problem here is there's now a timing change where it'll always take and then drop the lock before allocating directly, when it would skip the lock before.  Maybe that'll exercise new code paths.

Reply:
This code isn't changing any behavior. It's still skipping the lookaside list when we have either random allocations or custom allocation index dropping enabled. It sets the value to `NULL` here, which forces the code to call alloc below. And I'd rather not change the code once more if it's only for a comment. 😄